### ISSUE TITLE
Autocomplete stops working after importing "net" or "net/http"

### ISSUE THREAD

**Issue by hojgr** [fatih/vim-go#442]
Hello,

i've been having issues with autocomplete feature. With this code
package main

package main

```go
import "fmt"

func main() {
	fmt.Println("Hello world")

}
```

Autocomplete works fine. After I write `fmt.` all fmt options pop up. But after I import package "net", nothing works anymore.

```go
package main

import "fmt"
import "net"

func main() {
	fmt.Println("Hello world")
}
```

With this code, if I do `fmt.` or `net.`, nothing shows up. Also, in the "notification area" on bottom I see `-- User defined completion (^U^N^P) Pattern not found`

Am I writing Go wrong way?

**Comment by fatih** [comment]
You should use it without any problem. Here is my own setup with latest vim-go:

![image](https://cloud.githubusercontent.com/assets/438920/7821306/90bebbdc-03f6-11e5-8db4-ee4628de5042.png)

**Comment by hojgr** [comment]
Well, doesn't work for me unfortunetly. I'm new to Go and I think I've found the issue. I have my GOPATH in ~/go where I have bin, pkg, src. I tought I'd use `src` for my code and after installing all the tools I had some code in ~/go/src I didn't really want. So I removed it.

That is the reason to not work, right? Do I put all my code somewhere else than $GOPATH?

Thanks

**Comment by fatih** [comment]
Your code needs to be under $GOPATH. If you have it under that path everyone is ok. Example:

Assume GOPATH is `~/go`. Then your code should be under `~/go/src/`. If you don't put your code under $GOPATH vim-go will not work properly. Be sure everything is working fine.

Second completion for third party packages only works if the `pkg` folder exists and the packages are installed via `go install`.

**Comment by hojgr** [comment]
My code is in ~/go/src

It appears to be something with "gocode" (or my environment + gocode), after doing
`~/go/src $ gocode -f=json autocomplete main.go 8`
Autocomplete hangs and doesn't do anything.

**Comment by fatih** [comment]
@hojgr can you please try open the issue to gocode? I honestly don't know what to say. Another way of debugging it, open `gocode` in debug mode and start to use vim-go. You should see some output. 

Here is how to do: https://github.com/nsf/gocode#debugging

**Comment by hojgr** [comment]
It appears that I have installed everything the wrong way (using apt-get to get Go and some other tools...)

Prior to this issue, :GoInstallBinaries didn't work either (printed `go get` with some options, ...)

I have removed Go from my computer, tried to remove everything possible, read Go's installation guide and I'm trying to get it working from scratch. :GoInstallBinaries is working now and it's even way faster than when I was doing it manually.

I'll close it because it's not vim-go issue and it's maybe not even an issue at all. Thanks for help.
